sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/DoubleCounterSdk.java

```diff
@@ -16,7 +16,7 @@
 
   private DoubleCounterSdk(
       InstrumentDescriptor descriptor, InstrumentAccumulator instrumentAccumulator) {
-    super(descriptor, instrumentAccumulator);
+    super(descriptor, instrumentAccumulator, BoundInstrument::new);
```

Comment:
If all the implementations are the same, can we just inline this into the AbstractSynchronousInstrument constructor and call it a day? Are there ever likely to be other implementations for creating a bound instrument? Or is this the only way to cleanly get the type capture to work right?

Reply:
They are not the same. BoundInstrument is a different sub-class in the parent caller class

Reply:
oh, you're right. I totally missed that in the github view. That's what I get for not pulling down the code and looking at it fully. 👍

Reply:
Thanks :)